### ISSUE TITLE
Is multiple rank:multiple GPU supported? 

### ISSUE THREAD

**Issue by suranap** [ROCmSoftwarePlatform/MIOpen#2849]
I am running a small test ML training application using [FlexFlow](https://github.com/flexflow/FlexFlow) on [Frontier](https://docs.olcf.ornl.gov/systems/frontier_user_guide.html#frontier-compute-nodes). When I run 1 process using 8 GPUs, it takes 40min. When I run 8 processes, each using 1 GPU, it only takes 6 min. 

I turned off the cache using MIOPEN_DISABLE_CACHE. When I enable logging I can see MIOpen doesn't access the filesystem and it spends a lot of time in initialization. When I poke into the process with gdb, I see there are several threads doing MIOpen things. 

Is MIOpen able to run with 1 process accessing all GPUs on a node? If so, are there some recommended steps to debug why init takes so long?

**Comment by atamazov** [comment]
@suranap 
> I turned off the cache using MIOPEN_DISABLE_CACHE.

Disabling the persistent binary (kernel) cache enforces MIOpen to build the kernels more often. I do not quite understand the objective of switching it off. 

> I see there are several threads doing MIOpen things.

MIOpen does not spawn new threads/processes (except when it build kernels _offline_, e.g. with amdclang, but this is OFF by default), however HIP runtime or COMgr/hipRTC (_online_ compilers) can do that for their own internal needs.

> When I run 8 processes, each using 1 GPU, it only takes 6 min.
...
> Is MIOpen able to run with 1 process accessing all GPUs on a node? 

Can you please elaborate the goal of this? Why running 8 processes does not suit your needs?

**Comment by suranap** [comment]
(Sorry for the delay, I turned off notifications.) 

I'm using a parallel task runtime that could take a lot of memory. My node has 512G of system RAM. Slicing it over 8 processes (1 per gpu) leaves the runtime with 64G each. On nodes with Nvidia I'm able to run 1 process with 4 gpus. 

My program runs quickly on ROCm with 1 rank=1 gpu and on Nvidia when 1 rank=1 node. It runs much, much slower on ROCm with 1 rank=1 node. My profiler suggests it's stuck in MIOpen. What is causing this discrepancy? 

1. It could be the filesystem, so I disabled the cache. It's not the problem. 
2. It could be that MIOpen is only tested as 1 rank=1 gpu and other configurations are not supported. 
3. Or, you know MIOpen works 1 rank=N gpus. If so, then I need to further debug my program to find the cause. 

Which is it? Thanks.
